### DISCUSSION TITLE
Disabling force computation during training

### DISCUSSION THREAD

**Discussion by emaximova** [ACEsuit/mace#386] [Q&A] [answered]
Hi, 

We are interested in using MACE for predicting DFT energies of the molecules we are working with, particularly peptides. Since our focus is solely on energy prediction, our dataset does not include force labels. However, we could not figure out a straightforward way to disable the computation of forces during the training process. We have tried to set the --compute_forces flag to False, along with using --forces_weight=0.0 and --scaling="no_scaling" flags as per documentation examples, but the program continues to compute and predict forces, even without labels. 

I have also seen a discussion here regarding ignored flags and that the flags must obey a certain order, but this did not help to solve the issue. Could you please advise us on the correct procedure to disable this option?

Thank you very much,
Ekaterina

**Comment by gabor1** [comment]
HM..based on my experience you will get much better energy predictions if you included forces in your training. Even at a lower level of theory.

**Reply by gabor1** [reply]
Can you explain why you don't have forces available in your training data? It's typically free in electronic structure codes

**Reply by emaximova** [reply]
Thank you for your reply. In short, the software suite that our group works with and develops leverages quantum chemistry packages code for energy calculations and generates the training data we intend to use. Currently, we have not incorporated support for force calculations, as our applications do not require forces. However, considering your suggestions, we should probably add this support and regenerate the training data.

**Reply by gabor1** [reply]
Obviously it depends on how much effort it takes. But I suggest that you seriously consider it. For example, many quantum chemistry packages don't compute gradients for wave function methods such as coupled cluster. In this case, I suggest that you train your model in two stages, first you train to Hartree-Fock with forces (the HF calculation is the first step of a CC calculation anyway), and then in the second stage you continue fine-tuning the model using the CC energies. You will get a much better model than if you trained to just CC energies from scratch.

**Comment by ilyes319** [comment, marked as answer]
Dear @emaximova,
The model will predict the forces but it will get ignorded in the loss, so it will not affect your training. Just make sure to `--scaling="no_scaling"`.

As Gabor said, we highly recommend using forces during training.

**Comment by lamthuy** [comment]
In the forward function of MACE model you can disable force computation by setting:
```
compute_force: bool = False,
```
I guess it will ignore the calculation of forces. Since force calculation require gradient evaluation, if you don't need it you should skip the computation to save time.
